### DISCUSSION TITLE
Is it possible to reset the coder state and remove created files and opened tabs?

### DISCUSSION THREAD

**Discussion by mikhail-g** [coder/code-server#6578] [Q&A]
I want to use Coder for testing a vscode extension and for this purpose I need it to start with the default state before each test run. Is there a way to reset everything created in a coder session?

**Comment by code-asher** [comment]
Is this about code-server or Coder?  And if this is about Coder, could you expand a little more?

**Reply by code-asher** [reply]
If this is about code-server, then the opened tab state is stored in the browser, so you would have to clear your browser cache to delete that.  For created files, you would just have to delete them manually as part of your test teardown (or setup).

**Reply by mikhail-g** [reply]
It's about code-server. Thank you, opened tabs are removed now as expected. 
But for created files I still need to find some better solution then just removing them manually. I have tried copying them to the code-server container with `docker exec` but code-server loses access to them after they have been replaced with another version of the workspace files. The same happens with settings.json file and keybinding.json

**Reply by code-asher** [reply]
Hmm, I might need to see the test code to fully understand.  

When files like `settings.json` and `keybinding.json` are modified, the changes are supposed to be picked up automatically by code-server, I think.  

Losing access to those files sounds potentially like a permissions issue.  Are the permissions correct after the `docker exec`?
